recipes/perl-number-format/meta.yaml

```
@@ -0,0 +1,41 @@
+{% set cpan_name = "Number-Format" %}
+{% set version = "1.76" %}
+{% set sha256 = "0e0060eb363635a885706c6a26f5fcaafeae759f7b2acae49dda70e195dd44d6" %}
+
+package:
+  name: perl-{{ cpan_name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://cpan.metacpan.org/authors/id/R/RJ/RJBS/Number-Format-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  noarch: generic
+  script:
+    - perl Makefile.PL INSTALLDIRS=vendor NO_PERLLOCAL=1 NO_PACKLIST=1
+    - make && make test && make install VERBINST=1  # [not win]
+    - gmake && gmake test && gmake install VERBINST=1  # [win]
+
+requirements:
+  build:
+    - make
```

Have you tried the m2sys make?

```suggestion
  script:
    - perl Makefile.PL INSTALLDIRS=vendor NO_PERLLOCAL=1 NO_PACKLIST=1
    - make && make test && make install VERBINST=1

requirements:
  build:
    - make     # [not win]
    - m2-make  # [win]
```

---

Yes, actually that was what the first commit used...the CI just hung at the first `make` call, for several hours until I eventually pushed another commit and it canceled.

---

There are no CI logs for a run with `m2-make` for Windows. Can you point me to this? The first commit simply shows `make` without using the Cygwin (POSIX safe) version for Windows.

---

> There are no CI logs for a run with `m2-make` for Windows. Can you point me to this? The first commit simply shows `make` without using the Cygwin (POSIX safe) version for Windows.

My apologies, that was actually for the original batch of recipes I tried for PR #25136. The first couple of commits all used `m2-make` like your suggestion, but the Windows build for the third commit just hung. I then decided, rather than trying to trouble-shoot a multi-recipe PR which I assume would eat additional CI time, to open a new PR with a pure-Perl package that in theory should be simple to build and use that to troubleshoot the Windows build.

---

I just pushed a commit using `m2-make` and it has hung at the same point it did previously in the other PR.

---

Since I'm not sure if canceled CI jobs can be seen (I couldn't see the last one), the tail end of the log prior to the job freezing was:
```
(base) %SRC_DIR%>set "VS_YEAR=2015" 

(base) %SRC_DIR%>set "CMAKE_GENERATOR=Visual Studio 14 2015 Win64" 

(base) %SRC_DIR%>set "MSYS2_ARG_CONV_EXCL=/AI;/AL;/OUT;/out" 

(base) %SRC_DIR%>set "MSYS2_ENV_CONV_EXCL=CL" 

(base) %SRC_DIR%>call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64 
The system cannot find the path specified.

(base) %SRC_DIR%>set "INCLUDE=%PREFIX%\Library\include;" 

(base) %SRC_DIR%>set "LIB=%PREFIX%\Library\lib;" 

(base) %SRC_DIR%>call "C:\Miniforge\Scripts\..\condabin\conda_hook.bat" 

(base) %SRC_DIR%>call "C:\Miniforge\Scripts\..\condabin\conda.bat" activate "%PREFIX%" 
Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Number::Format
Writing MYMETA.yml and MYMETA.json
Microsoft Windows [Version 10.0.20348.2227]
(c) Microsoft Corporation. All rights reserved.
```
These lines:
```
Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Number::Format
Writing MYMETA.yml and MYMETA.json
```
I recognize as coming from the `perl Makefile.PL` command. After that should come the call to `make`. Not sure what the last few Microsoft copyright statements are, as I thought `m2-make` installed an MSYS version of GNU make.